x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/StoreKeyConfig.java

```diff
@@ -154,6 +155,26 @@ X509ExtendedTrustManager createTrustManager(@Nullable Environment environment) {
         }
     }
 
+    public List<Tuple<PrivateKey, X509Certificate>> getPrivateKeyEntries(Environment environment) {
```

Comment:
Is this method necessary (as well as making the class public, and using `instanceof` in the transport handler)?
I've found https://github.com/elastic/elasticsearch/blob/0220dfb3fe4de4ca416918a76a6fdcff0369f624/libs/ssl-config/src/test/java/org/elasticsearch/common/ssl/StoreKeyConfigTests.java#L138 that appears to do the same thing. Have you looked at that?

Reply:
The instanceof is there so that we know that this key manager was built from a keystore and not from PEM files. The idea was that the enrollment process depends on our self-generated key material and is not supposed to be an one-solution-fits-all thing, and this was an additional check to that effect. 

It is not strictly needed though, I could remove this if making StoreKeyConfig public is something that you find bothersome. 

> that appears to do the same thing. Have you looked at that?

I don't think you can iterate all the PrivateKeyEntries from the KeyManager, we'd need to know the alias name. Which we can know because we will be generating the keystores, and if we only support these, then it's doable. I'll take a look, thanks

Reply:
I'll leave this as is @albertzaharovits as StoreConfig is moving to common/ssl only and it will be public there so I don't think there is any concern adding a helper method that does exactly what we want it to there. Let me know if you feel strongly otherwise